### PR TITLE
update np min to 122

### LINKED ISSUE #9047: Drop support for Numpy 1.21
## Feature request

As per:

https://github.com/numba/numba/wiki/Minutes_2023_06_27

And:

https://numpy.org/neps/nep-0029-deprecation_policy.html

We have elected to drop support for Numpy 1.21 with Numba 0.58.

### PR DESCRIPTION
**Pull request by esc**
Adoption of #9093 

Fixes #9047.

### CODE REVIEW

**Comment by esc** [comment]
Merged in `main` post merge of #9105

**Comment by gmarkall** [comment]
gpuci run tests

**Review by gmarkall — DISMISSED** [review]
Approved on the basis of #9093 already being approved and this is exactly the same plus a merge from `main`.

**Comment by gmarkall** [comment]
Seems there's a problem with the docs build, so moved back to "Waiting on author".

**Comment by gmarkall** [comment]
Ah, the problem is that `ufunc_db` no longer imports `numpy_version`.

File: numba/np/ufunc_db.py [hunk 1]
```diff
@@ -49,7 +49,6 @@ def _fill_ufunc_db(ufunc_db):
     # module.
     from numba.np import npyfuncs
     from numba.cpython import cmathimpl, mathimpl, numbers
-    from numba.np.numpy_support import numpy_version
```
Comment (gmarkall) [inline numba/np/ufunc_db.py:52]: I think this needs to stay now that it is used to check for NumPy 1.25.

**Comment by gmarkall** [comment]
gpuci run tests

File: numba/np/ufunc_db.py (hunk 1 above)
Comment (esc) [reply numba/np/ufunc_db.py]: fixed in 1aef98a6bfba22d5c04615769ef03d1d9d4915ac

**Comment by esc** [comment]
I ran this through the Build-Farm as `numba_smoketest_cpu_yaml_117` and it passed fine. 🍏

**Pull request merged**